devimint/src/cli.rs

```diff
@@ -253,6 +253,7 @@ pub async fn handle_command(cmd: Cmd, common_args: CommonArgs) -> Result<()> {
                 let task_group = task_group.clone();
                 async move {
                     let dev_fed = DevJitFed::new(&process_mgr, skip_setup)?;
+                    dev_fed.finalize(&process_mgr).await?;
```

Comment:
:/

This is going to prematurely wait for everything to completely, making things slower.

fedimint-cli is supposed to block and wait for the client lock to be released, precisely so that concurrent usage is OK. If that's broken, we should fix it. If another tool / internal code is trying to access the db, we should add the lock there as well.